packages/moonstone/VideoPlayer/VideoPlayer.js

```diff
@@ -766,13 +777,25 @@ const VideoPlayerBase = class extends React.Component {
 			this.calculateMaxComponentCount(leftCount, rightCount, childrenCount);
 		}
 
-		const {source} = this.props;
-		const {source: nextSource} = nextProps;
+		const {source, preloadSource} = this.props;
+		const {source: nextSource, preloadSource: nextPreloadSource} = nextProps;
+		this.isVideoPreloaded = false;
+
+		if (preloadSource && compareSources(preloadSource, nextSource)) {
+			this.preloadSourcePlaying = !this.preloadSourcePlaying;
+			const currentVideoSource = this.video;
+			this.video = this.preloadVideo;
+			this.preloadVideo = currentVideoSource;
+			this.isVideoPreloaded = true;
+		}
+
+		if (nextPreloadSource) {
```

Comment:
Do we need to compare against the previous preload source?  This will happen every time, will it not?